From 9fb3de8e9268e8e81128fc77328495567140e082 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Sat, 25 May 2013 23:57:04 -0400
Subject: [PATCH] configure, secur32: In gnutls-3.2, gnutls_mac_get_key_size()
 moved to crypto.h

Based on a patch by Ben Kohler <bkohler@gmail.com>

https://bugs.gentoo.org/show_bug.cgi?id=471077
http://bugs.winehq.org/show_bug.cgi?id=33649
---
 configure.ac                   | 3 ++-
 dlls/secur32/schannel_gnutls.c | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1941be1..2c10734 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1344,7 +1344,8 @@ then
     ac_save_CPPFLAGS="$CPPFLAGS"
     WINE_PACKAGE_FLAGS(GNUTLS,[gnutls])
     AC_CHECK_HEADER(gnutls/gnutls.h,
-        [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <gnutls/gnutls.h>]], [[static typeof(gnutls_mac_get_key_size) *func; if (func) return 0;]])],
+        [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <gnutls/gnutls.h>
+#include <gnutls/crypto.h>]], [[static typeof(gnutls_mac_get_key_size) *func; if (func) return 0;]])],
             [WINE_CHECK_SONAME(gnutls,gnutls_global_init,,[GNUTLS_CFLAGS=""],[$GNUTLS_LIBS])])],
         [GNUTLS_CFLAGS=""])
     CPPFLAGS="$ac_save_CPPFLAGS"
diff --git a/dlls/secur32/schannel_gnutls.c b/dlls/secur32/schannel_gnutls.c
index 70b8276..cbe64e8 100644
--- a/dlls/secur32/schannel_gnutls.c
+++ b/dlls/secur32/schannel_gnutls.c
@@ -25,6 +25,7 @@
 #include <stdarg.h>
 #ifdef SONAME_LIBGNUTLS
 #include <gnutls/gnutls.h>
+#include <gnutls/crypto.h>
 #endif
 
 #include "windef.h"
-- 
1.8.2.1

